### PR TITLE
add default ServiceAccount needed by capabilities controller to the package

### LINKED ISSUE #3856: Add missing default ServiceAccount needed by capabilities controller
**Bug description**

Capabilities controller uses a default ServiceAccount for querying gvr, and this default ServiceAccount is missing in the capabilities package and needs to be added

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- (x) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by yharish991**
### What this PR does / why we need it
Capabilities controller requires a default ServiceAccount to query for GVR, this pr adds that missing default ServiceAccount to the capabilities package

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/3856

### Describe testing done for PR

Installed the capabilities package and verified that the capabilities controller is successfully querying for GVR.

### Release note

```release-note
None
```

### CODE REVIEW

File: packages/capabilities/bundle/config/upstream/rbac.yaml [hunk 1]
```diff
@@ -67,30 +67,11 @@ rules:
...
-      - get
-      - list
-      - watch
   - apiGroups:
       - ""
     resources:
-      - configmaps
-      - namespaces
-      - nodes
+      - serviceaccounts
```
[the remark this answers is not in the document]
Comment (danniel1205) [reply packages/capabilities/bundle/config/upstream/rbac.yaml:73]: question: do we need to get/list all serviceaccounts ? Or `resourceNames: tanzu-capabilities-manager-default-sa` is sufficient ?

File: packages/capabilities/bundle/config/upstream/rbac.yaml [hunk 2]
```diff
@@ -67,30 +67,11 @@ rules:
...
+      - secrets
```
Comment (danniel1205) [inline packages/capabilities/bundle/config/upstream/rbac.yaml:74]: question: why do we need get/list secrets now ?

File: packages/capabilities/bundle/config/upstream/rbac.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yharish991) [reply packages/capabilities/bundle/config/upstream/rbac.yaml:73]: yes, in the [capability spec](https://github.com/vmware-tanzu/tanzu-framework/blob/main/apis/core/v1alpha1/capabilities_types.go#L19) users can provide service account name and query for resources they have access to. Check this security model for capabilities https://github.com/vmware-tanzu/tanzu-framework/blob/main/docs/api-machinery/capability-discovery.md#security-model to get a better idea of why we accept service account in capability spec

File: packages/capabilities/bundle/config/upstream/rbac.yaml (hunk 2 above)
Comment (yharish991) [reply packages/capabilities/bundle/config/upstream/rbac.yaml:74]: the security model for capabilities allows users to provide service account name to query resources they have access to, in order to achieve that, we build a custom rest.Config for each custom resource and use that to query resources https://github.com/vmware-tanzu/tanzu-framework/blob/main/capabilities/controller/pkg/capabilities/core/capability_controller.go#L54

**Review by danniel1205 — APPROVED** [review]
lgtm, thanks for the change

**Pull request merged**